### ISSUE TITLE
Squiz.Scope.StaticThisUsage incorrectly looking inside anon classes

### ISSUE THREAD

**Issue by pajon** [squizlabs/PHP_CodeSniffer#1776]
Hello, 

there is false positive error in StaticThisUsageSniff. Anonymous class definition inside static method with using $this variable, the sniff consider it as mistake.

Example code:
```
private static function functionExample(array $data)
{
    return new class()
    {
        private $data;

        public function __construct(array $data)
        {
            $this->data = $data;
        }
    };
}
```

Result:
`Usage of "$this" in static methods will cause runtime errors`

**Comment by gsherwood** [comment]
Thanks for reporting this. The sniff now skips anon classes.
